### PR TITLE
storage: inode size configurable when constructing XFS filesystem

### PR DESCRIPTION
**Pull request by sahid**
Related to:
  https://bugs.launchpad.net/charm-swift-storage/+bug/1826552

Signed-off-by: Sahid Orentino Ferdjaoui <sahid.ferdjaoui@canonical.com>

### CODE REVIEW

File: charmhelpers/contrib/storage/linux/utils.py [hunk 1]
```diff
@@ -110,17 +110,20 @@ def is_device_mounted(device):
...
+    :ptype: force: boolean
+    :param inode_size: The inode size in bytes, min 256, max 2048.
+    :ptype: inode_size: integer
+    """
     cmd = ['mkfs.xfs']
     if force:
         cmd.append("-f")
 
-    cmd += ['-i', 'size=1024', device]
+    cmd += ['-i', 'size=%d' % inode_size, device]
```
Comment (ajkavanagh) [inline charmhelpers/contrib/storage/linux/utils.py]: Minor quibble, in that I prefer `.format(...)` over `%` but not enough to nack the review!

**Comment by tytus-kurek** [comment]
I don't mind preferring one over the other. [#303](https://github.com/juju/charm-helpers/pull/313) has been already tested together with https://review.opendev.org/#/c/656265/ , but I can run another test once the swift-storage charm is updated with the patched charm-helpers.

**Comment by sahid** [comment]
Thanks for your review Alex. I marked the bug assigned to Tytus. We can consider my PR abandoned.

Tytus, as indicated in the launchpad bug, please consider to mark yourself on "assigned to" and don't compete with somebody which has already indicated himself working on it :-)

Thanks,
s.

**Comment by tytus-kurek** [comment]
Sahid,

There was no intention to compete with anyone. I had the patches ready for submitting on Friday afternoon, just hadn't tested them yet. Today after testing I submitted them and then it turned out we've been working on that in parallel.

One more time, apologies for not marking myself as an assignee before starting an actual work.

**Pull request closed without merging**